examples/llama3-vllm-gcp/llama3_vllm_gcp.py

```diff
@@ -0,0 +1,188 @@
+# The actual vLLM imports can happen within the functions that will be sent to the Runhouse cluster via an env.
+
+import asyncio
+
+import runhouse as rh
+
+# Next, we define a class that will hold the model and allow us to send prompts to it.
+# You'll notice this class inherits from `rh.Module`.
+# This is a Runhouse class that allows you to run code in your class on a remote machine.
+#
+# Learn more in the [Runhouse docs on functions and modules](/docs/tutorials/api-modules).
+class LlamaModel(rh.Module):
+    def __init__(self, model_id="meta-llama/Meta-Llama-3-8B-Instruct", **model_kwargs):
+        super().__init__()
+        self.model_id, self.model_kwargs = model_id, model_kwargs
+        self.engine = None
+
+    def load_engine(self):
+        from vllm.engine.arg_utils import AsyncEngineArgs
+        from vllm.engine.async_llm_engine import AsyncLLMEngine
+
+        args = AsyncEngineArgs(
+            model=self.model_id,  # Hugging Face Model ID
+            tensor_parallel_size=1,  # Increase if using additional GPUs
+            trust_remote_code=True,  # Trust remote code from Hugging Face
+            enforce_eager=True,  # Set to False for production use cases
+        )
+        self.engine = AsyncLLMEngine.from_engine_args(args)
+
+    async def generate(self, prompt: str, **sampling_params):
+        from vllm.sampling_params import SamplingParams
+        from vllm.utils import random_uuid
+
+        if not self.engine:
+            self.load_engine()
+
+        sampling_params = SamplingParams(**sampling_params)
+        request_id = random_uuid()
+        results_generator = self.engine.generate(prompt, sampling_params, request_id)
+
+        async for output in results_generator:
+            final_output = output
+        responses = []
+        for output in final_output.outputs:
+            responses.append(output.text)
+        return responses
+
+
+# ## Set up Runhouse primitives
+#
+# Now, we define the main function that will run locally when we run this script and set up
+# our Runhouse module on a remote cluster. First, we create a cluster with the desired instance type and provider.
+# Our `instance_type` here is defined as `L4:1`, which is the accelerator type and count that we need. We could
+# alternatively specify a specific [GCP instance](https://cloud.google.com/compute/docs/gpus) type, such as `g2-standard-8`.
+#
+# Learn more in the [Runhouse docs on clusters](/docs/tutorials/api-clusters).
+#
+# :::note{.info title="Note"}
+# The Python code we'll run is contained in an asynchronous function, `main`. To make this guide more readable, it's
+# contents are rendered as top-level code snippets, but they should be included in the `main` method for running.
+# :::
+async def main():
+    gpu_cluster = rh.cluster(
+        name="rh-l4x",
+        instance_type="L4:1",
+        memory="32+",
+        provider="gcp",
+        # (Optional) Include the following to create exposed TLS endpoints
```

Comment:
not sure we've been doing this for other examples, but thoughts on adding a note on how to incorporate autostop? been getting a lot of questions about that lately

Reply:
Oh yeah... I definitely think we should do that! Maybe including it here in the cluster AND also at the end of the example explain how to use `sky status` `sky down`?

Reply:
i think showing how to set the `autostop` field in the factory would probably suffice for now, but if it doesn't feel too cluttered to add the sky commands / link to the docs that describe them that that could work too

Reply:
Made an update with both just because I'm not sure it's super easy to find elsewhere - lmk if that seems good!